### PR TITLE
Adding test coverage around rh_cloud stubbed cases

### PR DESCRIPTION
**Pull request by adarshdubey-star**
**Test results 6.10**

```
pytest tests/foreman/api/test_rhcloud_inventory.py -k test_rhcloud_inventory_api_e2e
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.6.1, reportportal-5.0.8, forked-1.3.0, ibutsu-1.16, cov-3.0.0, xdist-2.4.0
collected 8 items / 7 deselected / 1 selected                                                                                                                                                                     

tests/foreman/api/test_rhcloud_inventory.py .                                                                                                                                                               [100%]

============================================================================ 1 passed, 7 deselected,in 1211.83s (0:20:11) ============================================================================



pytest tests/foreman/api/test_rhcloud_inventory.py -k test_rh_cloud_tag_values
=============================================================================================== test session starts ===============================================================================================
platform linux -- Python 3.8.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/addubey/work/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, mock-3.6.1, reportportal-5.0.8, forked-1.3.0, ibutsu-1.16, cov-3.0.0, xdist-2.4.0
collected 8 items / 7 deselected / 1 selected                                                                                                                                                                     

tests/foreman/api/test_rhcloud_inventory.py .                                                                                                                                                               [100%]

============================================================================ 1 passed, 7 deselected,in 1083.83s (0:18:03) =====================================================================

```

### CODE REVIEW

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 1]
```diff
@@ -78,24 +80,41 @@ def test_rhcloud_inventory_api_e2e(
...
+    wait_for(
+        lambda: rhcloud_sat_host.api.ForemanTask()
+        .search(query={'search': f'{generate_report_task} and started_at >= "{timestamp}"'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:106]:
```suggestion
generate_inventory_report(rhcloud_sat_host, org)
```

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 2]
```diff
@@ -78,24 +80,41 @@ def test_rhcloud_inventory_api_e2e(
...
     :CaseImportance: Critical
 
-    :BZ: 1807829, 1926100, 1965234
+    :BZ: 1807829, 1926100, 1965234, 1824183, 1879453
     """
     org, ak = organization_ak_setup
     virtual_host, baremetal_host = rhcloud_registered_hosts
     local_report_path = robottelo_tmp_dir.joinpath(f'{gen_alphanumeric()}_{org.id}.tar.xz')
     # Generate report
-    generate_inventory_report(rhcloud_sat_host, org)
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:91]: @adarshdubey-star No need to remove `generate_inventory_report(rhcloud_sat_host, org)`, it does the same thing for which you have added new code.

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 3]
```diff
@@ -78,24 +80,41 @@ def test_rhcloud_inventory_api_e2e(
...
     # Download report
     rhcloud_sat_host.api.Organization(id=org.id).rh_cloud_download_report(
         destination=local_report_path
     )
     common_assertion(local_report_path)
     # Assert Hostnames, IP addresses, and installed packages are present in report.
     json_data = get_report_data(local_report_path)
+    json_meta_data = get_report_metadata(local_report_path)
+
+    assert json_meta_data['source_metadata'] == {'foreman_rh_cloud_version': '4.0.27'}
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:116]: @adarshdubey-star Can you get the `tfm-rubygem-foreman_rh_cloud` package version from satellite and compare it with `json_meta_data['source_metadata']['foreman_rh_cloud_version']`?

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 4]
```diff
@@ -334,3 +353,65 @@ def test_include_parameter_tags_setting(
...
+    wait_for(
+        lambda: rhcloud_sat_host.api.ForemanTask()
+        .search(query={'search': f'{generate_report_task} and started_at >= "{timestamp}"'})[0]
+        .result
+        == 'success',
+        timeout=400,
+        delay=15,
+        silent_failure=True,
+        handle_exception=True,
+    )
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:406]: you can use `generate_inventory_report(rhcloud_sat_host, org)` here.

File: tests/foreman/api/test_rhcloud_inventory.py (hunk 3 above)
Comment (jameerpathan111) [reply tests/foreman/api/test_rhcloud_inventory.py:116]: Also, I guess you forgot to add assertion for `infrastructure_type`.

Comment (adarshdubey-star) [reply tests/foreman/api/test_rhcloud_inventory.py]: Noted!

File: tests/foreman/api/test_rhcloud_inventory.py (hunk 4 above)
Comment (adarshdubey-star) [reply tests/foreman/api/test_rhcloud_inventory.py]: Got it!

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 5]
```diff
@@ -63,7 +65,11 @@ def common_assertion(report_path):
 @pytest.mark.run_in_one_thread
 @pytest.mark.tier3
 def test_rhcloud_inventory_api_e2e(
-    inventory_settings, organization_ak_setup, rhcloud_registered_hosts, rhcloud_sat_host
+    inventory_settings,
+    organization_ak_setup,
+    rhcloud_registered_hosts,
+    rhcloud_sat_host,
+    default_sat,
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:72]:
```suggestion
```
added `default_sat` here by mistake?

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 6]
```diff
@@ -96,6 +104,11 @@ def test_rhcloud_inventory_api_e2e(
     common_assertion(local_report_path)
     # Assert Hostnames, IP addresses, and installed packages are present in report.
     json_data = get_report_data(local_report_path)
+    json_meta_data = get_report_metadata(local_report_path)
+    package_version = default_sat.run('rpm -qa --qf "%{VERSION}" tfm-rubygem-foreman_rh_cloud')
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:108]:
```suggestion
    package_version = rhcloud_sat_host.run('rpm -qa --qf "%{VERSION}" tfm-rubygem-foreman_rh_cloud')
```

Comment (adarshdubey-star) [reply tests/foreman/api/test_rhcloud_inventory.py:108]: Yes, will update this.

**Review by jameerpathan111 — APPROVED** [review]

File: tests/foreman/api/test_rhcloud_inventory.py [hunk 7]
```diff
@@ -334,3 +349,54 @@ def test_include_parameter_tags_setting(
...
+
+@pytest.mark.tier3
+def test_rh_cloud_tag_values(
+    inventory_settings, organization_ak_setup, rhcloud_sat_host, rhcloud_registered_hosts
+):
+    """Verify that tag values are escaped properly when hostgroup name
+        contains " (double quote) in it.
+
+    :id: ea7cd7ca-4157-4aac-ad8e-e66b88740ce3
+
```
Comment (jameerpathan111) [inline tests/foreman/api/test_rhcloud_inventory.py:362]:
```suggestion
:customerscenario: true
```

Comment (adarshdubey-star) [reply tests/foreman/api/test_rhcloud_inventory.py:362]: updated!

**Comment by devendra104** [comment]
trigger: test-robottelo
pytest: -k test_rh_cloud_tag_values tests/foreman/api/test_rhcloud_inventory.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: -k test_rh_cloud_tag_values tests/foreman/api/test_rhcloud_inventory.py

**Comment by swadeley** [comment]
Hi

build 587 failed: `nailgun.entity_mixins.MissingValueError: Cannot find a value for the "environment" field`

**Review by pondrejk — APPROVED** [review]
ack pending test results

**Comment by swadeley** [comment]
Re the nailgun error https://github.com/SatelliteQE/nailgun/pull/799

**Pull request merged**